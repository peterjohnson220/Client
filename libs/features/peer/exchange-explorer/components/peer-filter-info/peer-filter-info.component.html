<div class="filter-info w-100">
  <label class="filter-info-org-label">Orgs: <span class="filter-info-span">{{orgCount != null ? orgCount.toLocaleString() : orgCount}}</span></label>
  <label class="filter-info-inc-label">Incs: <span class="filter-info-span">{{incCount != null ? incCount.toLocaleString() : incCount}}</span></label>
  <div class="reset-link filters-link"
       [class.disabled]="!hasSelections" (click)="handleClearFiltersLinkClicked()">
    Clear Filters
  </div>
</div>
