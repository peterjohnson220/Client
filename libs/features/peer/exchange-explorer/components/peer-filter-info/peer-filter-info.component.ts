import { Component, EventEmitter, Input, Output } from '@angular/core';

@Component({
  selector: 'pf-peer-filter-info',
  templateUrl: './peer-filter-info.component.html',
  styleUrls: ['./peer-filter-info.component.scss']
})

export class PeerFilterInfoComponent {
  @Input() orgCount: number;
  @Input() incCount: number;
  @Input() hasSelections: boolean;
  @Output() clearFiltersLinkClicked = new EventEmitter();

  constructor() { }

  handleClearFiltersLinkClicked() {
    if (this.hasSelections) {
      this.clearFiltersLinkClicked.emit();
    }
  }

}
