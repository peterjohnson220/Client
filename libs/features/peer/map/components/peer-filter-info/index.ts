export * from './peer-filter-info.component';
