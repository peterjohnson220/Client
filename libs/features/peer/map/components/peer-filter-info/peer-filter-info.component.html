<div class="filter-info">
  <label class="filter-info-org-label">Orgs: <span class="filter-info-span">{{orgCount != null ? orgCount.toLocaleString() : orgCount}}</span></label>
  <label class="filter-info-inc-label">Incs: <span class="filter-info-span">{{incCount != null ? incCount.toLocaleString() : incCount}}</span></label>
  <div class="reset-link filters-link" [class.disabled]="!hasSelections" (click)="handleResetFiltersLinkClicked()">
    Reset Filters
  </div>
</div>