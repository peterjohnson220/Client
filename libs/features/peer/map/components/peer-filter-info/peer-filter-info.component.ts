import { Component, EventEmitter, Input, Output } from '@angular/core';

@Component({
  selector: 'pf-peer-filter-info',
  templateUrl: './peer-filter-info.component.html',
  styleUrls: ['./peer-filter-info.component.scss']
})

export class PeerFilterInfoComponent {
  @Input() orgCount: number;
  @Input() incCount: number;
  @Input() hasSelections: boolean;
  @Input() showResetLink = true;
  @Output() resetAllFiltersClicked = new EventEmitter();

  constructor() { }

  handleResetFiltersLinkClicked() {
    this.resetAllFiltersClicked.emit();
  }

}
