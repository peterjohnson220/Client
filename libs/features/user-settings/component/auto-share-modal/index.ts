export * from './auto-share-modal.component';
