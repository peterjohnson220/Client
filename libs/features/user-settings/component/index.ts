export * from './auto-share-modal';
export * from './auto-shared-users';
export * from './shareable-user';
