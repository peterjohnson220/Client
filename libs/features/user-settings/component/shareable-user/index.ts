export * from './shareable-user.component';
