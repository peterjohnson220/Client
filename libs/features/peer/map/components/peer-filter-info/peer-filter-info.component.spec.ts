import { NO_ERRORS_SCHEMA } from '@angular/core';
import { TestBed, ComponentFixture } from '@angular/core/testing';

import spyOn = jest.spyOn;

import { PeerFilterInfoComponent } from './peer-filter-info.component';

describe('Feature - Peer - Pay Market Filter Info Component', () => {
  let fixture: ComponentFixture<PeerFilterInfoComponent>;
  let instance: PeerFilterInfoComponent;

  // Configure Testing Module before each test
  beforeEach(() => {
    TestBed.configureTestingModule( {
      declarations: [
        PeerFilterInfoComponent
      ],
      // Shallow Testing
      schemas: [ NO_ERRORS_SCHEMA ]
    });

    fixture = TestBed.createComponent(PeerFilterInfoComponent);
    instance = fixture.componentInstance;
  });

  it('should provide correct values to the pf-peer-filter-info component', () => {
    fixture.detectChanges();

    expect(fixture).toMatchSnapshot();
  });

  it( 'should emit a resetAllFiltersClicked event when handling ResetFiltersLinkClicked', () => {
    spyOn(instance.resetAllFiltersClicked, 'emit');

    instance.handleResetFiltersLinkClicked();

    expect(instance.resetAllFiltersClicked.emit).toHaveBeenCalled();
  });
});
