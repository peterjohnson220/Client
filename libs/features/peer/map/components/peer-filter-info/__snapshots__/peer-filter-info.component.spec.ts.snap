// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Feature - Peer - Pay Market Filter Info Component should provide correct values to the pf-peer-filter-info component 1`] = `
<pf-peer-filter-info
  resetAllFiltersClicked={[Function EventEmitter]}
>
  <div
    class="filter-info"
  >
    <label
      class="filter-info-org-label"
    >
      Orgs: 
      <span
        class="filter-info-span"
      >
        
      </span>
    </label>
    <label
      class="filter-info-inc-label"
    >
      Incs: 
      <span
        class="filter-info-span"
      >
        0
      </span>
    </label>
    <div
      class="reset-link filters-link disabled"
    >
       Reset Filters 
    </div>
  </div>
</pf-peer-filter-info>
`;
